pkg/azurefile/nodeserver.go

```diff
@@ -95,6 +96,35 @@ func (d *Driver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 				return nil, status.Errorf(codes.InvalidArgument, fmt.Sprintf("invalid mountPermissions %s", perm))
 			}
 		}
+
+		// get runtime class using pod info from volume context
+		runtimeClass, err := getRuntimeClassForPod(ctx, d.kubeconfig, context[podNameField], context[podNamespaceField])
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get runtime class for pod %s/%s: %v", context[podNamespaceField], context[podNameField], err)
+		}
+		klog.V(2).Infof("NodePublishVolume: volume(%s) mount on %s with runtimeClass %s", volumeID, target, runtimeClass)
+		if isConfidentialRuntimeClass(runtimeClass) {
+			klog.V(2).Infof("NodePublishVolume for volume(%s) where runtimeClass %s is kata-cc", volumeID, runtimeClass)
+			source := req.GetStagingTargetPath()
+			if len(source) == 0 {
+				return nil, status.Error(codes.InvalidArgument, "Staging target not provided")
+			}
+			// Load the mount info from staging area
+			mountInfo, err := directvolume.VolumeMountInfo(source)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to load mount info from %s: %v", source, err)
+			}
+			if mountInfo == nil {
+				return nil, status.Errorf(codes.Internal, "mount info is nil for volume %s", volumeID)
+			}
+			data, _ := json.Marshal(mountInfo)
+			err = directvolume.Add(target, string(data))
```

Comment:
Do you have a KEP for this CSI extension? How are 3rd party CSI driver vendors supposed to support kata mounts?

@andyzhangx in general, I'd appreciate if such experimental features were clearly marked as experimental, e.g. behind a feature gate and explicitly called out in docs.

Reply:
yes, we should add a feature flag(see my previous comment: https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/1971#discussion_r1674168844)  to enable/disable this feature, disable by default, and enable by default in e2e tests.

I think @arc9693 could answer this. not sure whether we have a public doc for this CSI extension, we only do that as internal discussion.

Reply:
I am working on this.